cypress/integration/chi-custom-elements/carousel.spec.js

```diff
@@ -216,4 +216,49 @@ describe('Carousel', () => {
       cy.get('@prevButton').should('be.disabled');
     });
   });
+
+  describe('Autoplay', () => {
+    beforeEach(() => {
+      cy.get("[data-cy='autoplay']").as('autoplay');
+      cy.get('@autoplay')
+        .find(`.${CAROUSEL_DOTS_CLASS}`)
+        .children()
+        .as('controls');
+    });
+
+    it('should contain prop autoplay and is visible', () => { 
+      cy.get('@autoplay').find(`.${CAROUSEL_CLASS}`).should('be.visible') 
+      cy.get('@autoplay').should('be.visible');
+    });
+
+    it('should have prop autoplay with interval', () => {
+      cy.get('@autoplay')
+        .find('.chi-carousel__item:first')
+        .contains('1');
+      cy.get('@autoplay')
+        .should('have.prop', 'autoplay', true)
+        .should('have.prop', 'interval', 5);
+      cy.wait(5)
+      cy.get('@autoplay')
+          .find('.chi-carousel__item:nth-child(4)')
+          .contains('4');
+    });
+
+    it('should hide buttons when no-button-controllers is true', () => {
+      cy.get('@autoplay')
```

Comment:
I think it's better to separate the `autoplay` tests from the `no-button-controllers` tests, what do you think @jllr?
By the way I would test in a separate test the property, the same as in the autoplay.

In terms of functionality you need to test if the **previous** and **next** control are not visible, and you are only testing one of them. Also, you don't need to wait 5 seconds, as this test is separate from the `autoplay`. 

You can find `CAROUSEL_NEXT_CLASS` and `CAROUSEL_PREVIOUS_CLASS` in the file to guide you in performing this test correctly.

Reply:
Agree on this, `autoplay` and `no-button-controllers` should be two different tests.